net/tailscale/files/tailscale.init

```diff
@@ -4,44 +4,251 @@
 # Copyright (C) 2021 CZ.NIC z.s.p.o. (https://www.nic.cz/)
 # SPDX-License-Identifier: Apache-2.0
 
+START=90
+
 USE_PROCD=1
-START=80
 
-start_service() {
-  local state_file
-  local port
-  local std_err std_out
+PROG=/usr/sbin/tailscale
+PROGD=/usr/sbin/tailscaled
+CONFIG_PATH=/var/lib/tailscale
 
-  config_load tailscale
-  config_get_bool std_out "settings" log_stdout 1
-  config_get_bool std_err "settings" log_stderr 1
-  config_get port "settings" port 41641
-  config_get state_file "settings" state_file /etc/tailscale/tailscaled.state
-  config_get fw_mode "settings" fw_mode nftables
+service_triggers() {
+  procd_add_reload_trigger "tailscale"
+  procd_add_interface_trigger "interface.*.up" wan /etc/init.d/tailscale reload
+}
 
-  /usr/sbin/tailscaled --cleanup
+section_enabled() {
+  config_get_bool enabled "$1" 'enabled' 0
+  [ $enabled -gt 0 ]
+}
+
+custom_instance() {
+  local cfg="$1"
+  local accept_routes hostname accept_dns advertise_exit_node exit_node advertise_routes site_to_site subnet_routes flags login_server authkey std_out std_err
+  local ARGS=" up --reset"
+
+  config_get_bool accept_routes $cfg 'accept_routes'
+  config_get hostname $cfg 'hostname'
+  config_get_bool accept_dns $cfg 'accept_dns'
+  config_get_bool advertise_exit_node $cfg 'advertise_exit_node'
+  config_get exit_node $cfg 'exit_node'
+  config_get advertise_routes $cfg 'advertise_routes'
+  config_get_bool site_to_site $cfg 'site_to_site'
+  config_get flags $cfg 'flags'
+  config_get login_server $cfg 'login_server'
+  config_get authkey $cfg 'authkey'
+  config_get_bool std_out $cfg 'log_stdout'
+  config_get_bool std_err $cfg 'log_stderr'
+
+  [ "$accept_routes" = "1" ] && ARGS="$ARGS --accept-routes=true"
+  [ -n "$hostname" ] && ARGS="$ARGS --hostname=$hostname"
+  [ "$accept_dns" = "0" ] && ARGS="$ARGS --accept-dns=false"
+  [ "$advertise_exit_node" = "1" ] && ARGS="$ARGS --advertise-exit-node"
+  [ -n "$exit_node" ] && ARGS="$ARGS --exit-node=$exit_node --exit-node-allow-lan-access=true"
+  [ -n "$advertise_routes" ] && ARGS="$ARGS --advertise-routes=$(echo $advertise_routes | tr ' ' ',')"
+  [ "$site_to_site" = "1" ] && ARGS="$ARGS --snat-subnet-routes=false"
+  [ -n "$flags" ] && ARGS="$ARGS $flags"
+  [ -n "$login_server" ] && ARGS="$ARGS --login-server=$login_server"
+  [ -n "$authkey" ] && ARGS="$ARGS --authkey=$authkey"
 
   procd_open_instance
-  procd_set_param command /usr/sbin/tailscaled
+  procd_set_param command $PROG $ARGS
+  procd_set_param stdout "$std_out"
+  procd_set_param stderr "$std_err"
+  procd_close_instance
+  (
+    [ -f "/var/run/tailscale.wait.pid" ] && return
+    touch /var/run/tailscale.wait.pid
```

Comment:
Is this supposed to mimic `flock`?

Reply:
Then why not use `flock`? It seems to come as standard on OpenWrt as part of BusyBox:
```bash
flock --help
BusyBox v1.36.1 (2025-04-10 20:39:58 UTC) multi-call binary.

Usage: flock [-sxun] FD | { FILE [-c] PROG ARGS }

[Un]lock file descriptor, or lock FILE, run PROG

	-s	Shared lock
	-x	Exclusive lock (default)
	-u	Unlock FD
	-n	Fail rather than wait
```